core/arch/arm/crypto/sm4_armv8a_aese_a64.S

```diff
@@ -0,0 +1,1123 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) Hisilicon Technologies Co., Ltd. 2023. All rights reserved.
+ * Copyright (C) 2022, Alibaba Group.
+ * Copyright (C) 2022 Tianjia Zhang <tianjia.zhang@linux.alibaba.com>
+ *
+ * SM4 optimization for ARMv8 by NEON and AES HW instruction, which is an
+ * optional Cryptographic Extension for ARMv8-A.
+ *
+ * The NEON implementation refers to Linux kernel (sm4-neon-core.S contributed
+ * by Tianjia Zhang <tianjia.zhang@linux.alibaba.com>).
```

Comment:
The problem is, this code was contributed to Linux with a GPL-2.0-or-later license. It cannot be re-licensed as BSD-2-Clause here without the author's permission.

In practice what it means is, we need an Acked-by: tag from Tianjia Zhang for the BSD-2-Clause license.